Sources/NIO/RecvByteBufferAllocator.swift

```diff
@@ -55,7 +55,7 @@ public struct AdaptiveRecvByteBufferAllocator: RecvByteBufferAllocator {
     private var nextReceiveBufferSize: Int
     private var decreaseNow: Bool
 
-    private static let maximumAllocationSize = 1 << 31
+    private static let maximumAllocationSize = 1 << 30
```

Comment:
The maximum power of 2 that can be stored in an `Int32` is 2^30 not 2^31: `1 << 31` actually sets the sign bit and causes segfaults on Android armv7.

Reply:
I don't follow - where is the missing bit?

Reply:
Scratch that, got it